### DISCUSSION TITLE
Getting the date from part of filename

### DISCUSSION THREAD

**Discussion by alan-daniel** [SilentVoid13/Templater#648] [Help]
So I'm new to all of this and have what I'd imagine is a pretty basic question (I've tried searching but either can't figure out the best way to search or can't find any answers):

How can I get the date from a portion of the file's title? The title I'm trying to read from will always start with a date but then has unrelated text after the date, so the format of it is `YYYY-MM-DD - extra text`

I'm wanting to get the date from the title in order to auto-populate a different date format in a couple of places in the body of the file. Thanks for any help!

**Comment by welpdx** [comment]
#### Try this:
```js
<%*
var title = this.app.workspace.getActiveFile().basename
var match = title.match(/\d{4}\-(0?[1-9]|1[012])\-(0?[1-9]|[12][0-9]|3[01])/g)
var titleDate = match[0]
%>
```

To insert that date later on in the same document, do:
```js
<%* tR+= titleDate %>
```
#### Demo
![img](https://i.imgur.com/xmtj8hw.gif)

Note: Regex pattern from [stackoverflow](https://stackoverflow.com/a/22061879/14451841)

Hope that helps 

~Welp
